### PR TITLE
rework debugprint to ts

### PR DESCRIPTION
**Pull request by CrispusDH**

### CODE REVIEW

**Comment by CrispusDH** [comment]
@cnishina , could you take a look at this PR, please

**Comment by CrispusDH** [comment]
Reference #5061

**Comment by CrispusDH** [comment]
@cnishina any updates?

File: lib/frameworks/debugprint.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
-var util = require('util'),
-    Logger = require('../logger').Logger;
+import * as util from 'util';
+import {Logger} from '../logger';
+import {Runner} from '../runner';
```
Comment (cnishina) [inline lib/frameworks/debugprint.ts]: `import {RunResults} from '../taskRunner';

We might need to update the RunResults to be more flexible:
https://github.com/angular/protractor/blob/selenium4/lib/taskRunner.ts#L9

```
export interface RunResults {
  taskId?: number;
  specs?: Array<string>;
  ...
```

File: lib/frameworks/debugprint.ts [hunk 2]
```diff
@@ -11,11 +12,9 @@ var logger = new Logger('debugger');
  * @param {Array} specs Array of Directory Path Strings.
  * @return {Promise} Promise resolved with the test results
  */
-exports.run = (runner, specs) => {
+export const run = (runner: Runner, specs: Array<string>) => {
```
Comment (cnishina) [inline lib/frameworks/debugprint.ts]:
```
export const run = (runner: Runner, specs: Array<string>): Promise<RunResults> => {
```

File: lib/frameworks/debugprint.ts [hunk 3]
```diff
@@ -11,11 +12,9 @@ var logger = new Logger('debugger');
...
   return new Promise(resolve => {
-    logger.info('Resolved spec files: ' + util.inspect(specs));
-    resolve({
-      failedCount: 0
-    });
+    logger.info(`Resolved spec files: ${util.inspect(specs)}`);
+    resolve({failedCount: 0});
```
Comment (cnishina) [inline lib/frameworks/debugprint.ts]: Once the changes are made to `RunResults` you probably do not need to cast. If for some reason there are type errors, you might need to cast this to `resolve({failedCount: 0} as RunResults);`

**Comment by cnishina** [comment]
Hmmm I guess not. Awesome!

**Pull request merged**